src/test_utils.jl

```diff
@@ -68,30 +69,46 @@ end
 """
     FakeTemplate{DistributionEstimate, SingleOutput}()
 
-A [`Template`](@ref) whose [`Model`](@ref) will predict a univariate normal
-distribution (with zero mean and unit standard deviation) for each observation.
+A [`Template`](@ref) whose [`Model`](@ref) will accept real value variables to predict a
+univariate normal distribution (with zero mean and unit standard deviation) for each
+observation.
 """
 function FakeTemplate{DistributionEstimate, SingleOutput}()
     FakeTemplate{DistributionEstimate, SingleOutput}() do num_variates, inputs
         @assert(num_variates == 1, "$num_variates != 1")
-        inputs = NamedDimsArray{(:features, :observations)}(inputs)
+        inputs = _handle_inputs(inputs)
         return NoncentralT.(3.0, zeros(size(inputs, :observations)))
     end
 end
 
 """
     FakeTemplate{DistributionEstimate, MultiOutput}()
 
-A [`Template`](@ref) whose [`Model`](@ref) will predict a multivariate normal
-distribution (with zero-vector mean and identity covariance matrix) for each observation.
+A [`Template`](@ref) whose [`Model`](@ref) will accept real value variables to predict a
+multivariate normal distribution (with zero-vector mean and identity covariance matrix) for
+each observation.
 """
 function FakeTemplate{DistributionEstimate, MultiOutput}()
     FakeTemplate{DistributionEstimate, MultiOutput}() do num_variates, inputs
         std_dev = ones(num_variates)
+        inputs = _handle_inputs(inputs)
         return [Product(Normal.(0, std_dev)) for _ in 1:size(inputs, 2)]
     end
 end
 
+"""
+    _handle_inputs(inputs::AbstractMatrix)
+    _handle_inputs(inputs::AbstractVector{<:Sampleable})
+
+Process the inputs to `predict` appropriately depending on whether they are real valued or 
+distributions over input variables.
+"""
+function _handle_inputs(inputs::AbstractVector{<:Sampleable})
+    return NamedDimsArray{(:features, :observations)}(hcat([mean(inputs[i]) for i in 1:size(inputs, 1)]...))
+end
+
+_handle_inputs(inputs::AbstractMatrix) = NamedDimsArray{(:features, :observations)}(inputs)
```

Comment:
Implementing these two methods of `_handle_inputs` means extra `FakeTemplate` methods aren't needed for the two `PredictInputTrait` cases.  Means we keep the same number of `FakeTemplate` methods as before.